sdk/src/contexts/mpc.rs

```diff
@@ -16,48 +15,166 @@ pub trait MPCContext {
     fn create_network_delivery_wrapper<M>(
         &self,
-        mux: Arc<NetworkMultiplexer>,
-        party_index: PartyIndex,
+        mux: std::sync::Arc<crate::network::NetworkMultiplexer>,
+        party_index: crate::round_based::PartyIndex,
         task_hash: [u8; 32],
-        parties: BTreeMap<PartyIndex, subxt_core::ext::sp_core::ecdsa::Public>,
-    ) -> color_eyre::Result<
-        crate::network::round_based_compat::NetworkDeliveryWrapper<M>,
-        crate::Error,
-    >
+        parties: std::collections::BTreeMap<crate::round_based::PartyIndex, crate::subxt_core::ext::sp_core::ecdsa::Public>,
+    ) -> Result<crate::network::round_based_compat::NetworkDeliveryWrapper<M>, crate::Error>
     where
-        M: Clone
-            + Send
-            + Unpin
-            + 'static
-            + serde::Serialize
-            + serde::de::DeserializeOwned
-            + round_based::ProtocolMessage;
-
-    /// Gets the party index from the participants map
-    async fn get_party_index(&self) -> color_eyre::Result<PartyIndex, crate::Error>;
-
-    /// Gets the participants in the MPC protocol
+        M: Clone + Send + Unpin + 'static + crate::serde::Serialize + crate::serde::de::DeserializeOwned + crate::round_based::ProtocolMessage,
+    {
+        Ok(crate::network::round_based_compat::NetworkDeliveryWrapper::new(mux, party_index, task_hash, parties))
+    }
+
+    /// Retrieves the current party index
+    async fn get_party_index(
+        &self,
+    ) -> Result<crate::round_based::PartyIndex, crate::Error> {
+        Ok(self.get_party_index_and_operators().await?.0 as _)
+    }
+
+    /// Retrieves the current participants in the MPC protocol
     async fn get_participants(
         &self,
-        client: &subxt::OnlineClient<crate::clients::tangle::runtime::TangleConfig>,
-    ) -> color_eyre::Result<BTreeMap<PartyIndex, AccountId32>, crate::Error>;
+    ) -> Result<
+        std::collections::BTreeMap<crate::round_based::PartyIndex, crate::subxt::utils::AccountId32>,
+        crate::Error,
+    > {
+        Ok(self.get_party_index_and_operators().await?.1.into_iter().enumerate().map(|(i, (id, _))| (i as _, id)).collect())
+    }
+
+    /// Retrieves the current blueprint ID from the configuration
+    ///
+    /// # Errors
+    /// Returns an error if the blueprint ID is not found in the configuration
+    fn blueprint_id(&self) -> crate::color_eyre::Result<u64> {
+        self.config()
+            .protocol_specific
+            .tangle()
+            .map(|c| c.blueprint_id)
+            .map_err(|err| crate::color_eyre::Report::msg(format!("Blueprint ID not found in configuration: {err}")))
+    }
+
+    /// Retrieves the current call ID for this job
+    ///
+    /// # Errors
+    /// Returns an error if failed to retrieve the call ID from storage
+    #[deprecated(note = "Use `context.call_id` field instead. This function is highly prone to race conditions")]
```

Comment:
Why are we putting deprecated code in when we don't really have any blueprints live? Just clean things up, accelerate towards the goal.